open-banking-accelerator/internal-apis/internal-webapps/com.wso2.openbanking.accelerator.consent.endpoint/src/main/java/com/wso2/openbanking/accelerator/consent/endpoint/api/ConsentValidationEndpoint.java

```diff
@@ -0,0 +1,198 @@
+        JSONObject requestData;
+        Object requestDataObj;
+
+        if (IdentityCommonUtil.getConsentJWTPayloadValidatorConfigEnabled()) {
+            try {
+                IdentityCommonUtil.validateJWTSignatureWithPublicKey(payload, requestSignatureAlias);
+                requestData = JWTUtils.decodeRequestJWT(payload, "body");
+            } catch (OpenBankingException e) {
+                log.error("Error while validating JWT signature", e);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Error while validating JWT " +
+                        "signature");
+            } catch (ParseException e) {
+                log.error("Error while decoding validation JWT", e);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Error while decoding validation JWT");
+            }
+        } else {
+            try {
+                requestDataObj = new JSONParser(JSONParser.MODE_PERMISSIVE).parse(payload);
+            } catch (net.minidev.json.parser.ParseException e) {
+                log.error("Unable to parse the request payload", e);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Unable to parse the request payload");
+            }
+            if (!(requestDataObj instanceof JSONObject)) {
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, "Payload is not a JSON object");
+            } else {
+                requestData = (JSONObject) requestDataObj;
+            }
+        }
+
+        JSONObject requestHeaders = (JSONObject) requestData.get("headers");
+        Set<String> headerNames = requestHeaders.keySet();
+        Iterator<String> headersIterator = headerNames.iterator();
+        TreeMap<String, String> headersMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        while (headersIterator.hasNext()) {
+            String headerName = headersIterator.next();
+            headersMap.put(headerName, requestHeaders.getAsString(headerName));
+        }
+
+        JSONObject requestPayload = (JSONObject) requestData.get("body");
+        String requestPath = requestData.getAsString("electedResource");
+        String consentId = requestData.getAsString("consentId");
+        String userId = requestData.getAsString("userId");
+        String clientId = null;
+        if (requestData.containsKey("clientId")) {
+            clientId = requestData.getAsString("clientId");
+        }
+        Map<String, String> resourceParams = (Map<String, String>) requestData.get("resourceParams");
+
+        if (consentId == null) {
+            throw new ConsentException(ResponseStatus.BAD_REQUEST, "Consent Id is mandatory for consent validation");
+        }
+
+        try {
+            //Adding query parameters to the resource map
+            resourceParams = ConsentUtils.addQueryParametersToResourceParamMap(resourceParams);
+        } catch (URISyntaxException e) {
```

Comment:
havn't logged the exception

Reply:
fixed with ea8fc6ef449ca5c80e471dcf3749615b21da0d39